### PR TITLE
Update backup instructions

### LINKED ISSUE #479: Update backup instructions
Backup instructions [here](https://cert-manager.io/docs/tutorials/backup/) could be updated.

We could add instructions as to how to update with Velero as well as a note about the order of restore (i.e `Secrets` should be created before `cert-manager` `CR`s and any ingress resources using `ingress-shim` )

### PR DESCRIPTION
**Pull request by irbekrm**
I did some manual backup and restore testing (mostly with Velero on GKE) and added a few more notes to the backup doc.

The fact that the `Certificate`s for `Ingress` lose the correct owner reference during a restore and `cert-manager` cannot recreate it, seems problematic from backup and restore perspective, as there is no straightforward way to exclude the `Ingress` `Certificate`s only from a Velero backup. 

I added a recommendation to not restore `CertificateRequest`s.
I don't think that restoring them would necessarily cause failures as such, but with the new identity fields it seems like we should not restore them since the restorer's identity is likely going to be different than creator's and it might just cause confusion.
Also `cert-manager` cannot recreate their status after a restore correctly, so they end up having Ready condition set to False, which I think would also be confusing for users.
In all cases I could think of there should be no need to restore completed `CertificateRequest`s, but perhaps there are some edge cases where they are needed?

Some issues related to `cert-manager` backup and restore:

- [#2318](https://github.com/jetstack/cert-manager/issues/2318)
- [#2112](https://github.com/jetstack/cert-manager/issues/2112)
- #234
- #397 
- [#3612](https://github.com/jetstack/cert-manager/issues/3612)

Some of the issues above refer to the currently documented backup strategy via `kubectl get.. kubectl apply`- I have not looked into that much as a part of this PR.
I have not yet properly tested restoring from an etcd snapshot which I was planning- that might be a separate PR.

Fixes #479 

/kind cleanup

### CODE REVIEW

File: content/en/docs/tutorials/backup.md [hunk 1]
```diff
@@ -49,5 +57,114 @@ created above after installing cert-manager.
...
+This section refers to backing up and restoring 'all' Kubernetes resources in a
+cluster- including some `cert-manager` ones- for scenarios such as disaster
+recovery, cluster migration etc.
+
+Note that we have not done any extensive testing, only a few common scenarios-
+it is recommended to test the backup and restore strategy before putting it in place
+as there are likely to be some other edge cases.
+We would like to ensure that users can reliably backup and restore `cert-manager`
+resources using common tools- if you encounter any errors, please do open a
+GitHub issue or PR an update to this document.
```
Comment (wallrj) [inline content/en/docs/tutorials/backup.md:71]:
```suggestion
*Note*: We have tested this process on simple Kubernetes test clusters with a limited set of Kubernetes releases. To avoid data loss, please test both the backup and the restore strategy on your own cluster before depending upon it in production. If you encounter any errors, please open a GitHub issue or a PR to document variations on this process for different Kubernetes environments. 
```

I suggest something like that ^^  instead, but happy to go with the original version if you prefer.

File: content/en/docs/tutorials/backup.md [hunk 2]
```diff
@@ -49,5 +57,114 @@ created above after installing cert-manager.
...
+
+### Avoiding unnecessary certificate reissuance
+
+#### Order of restore
+
+If `cert-manager` does not find a Kubernetes Secret with an X.509 certificate
+for a `Certificate`, reissuance will be triggered. To avoid unnecessary
+reissuance after a restore, ensure that Secrets are restored before
+`Certificate`s. Similarly, Secrets should be restored before `Ingress`es if you
+are using [`ingress-shim`](../../usage/ingress/).
```
Comment (wallrj) [inline content/en/docs/tutorials/backup.md:81]:
```suggestion
If `cert-manager` does not find a Kubernetes `Secret` with an X.509 certificate
for a `Certificate`, reissuance will be triggered. To avoid unnecessary
reissuance after a restore, ensure that `Secret`s are restored before
`Certificate`s. Similarly, `Secret`s should be restored before `Ingress`es if you
are using [`ingress-shim`](../../usage/ingress/).
```

Maybe quote `Secret` for consistency with the quoting of other Kubernetes resource types.

File: content/en/docs/tutorials/backup.md [hunk 3]
```diff
@@ -49,5 +57,114 @@ created above after installing cert-manager.
...
+ephemeral resources (i.e a temporary Secret holding a private key), so
+`cert-manager` cannot always correctly re-create the status of these resources
+by just looking at the cluster state at some later time.
+
+In most cases backup and restore tools will not restore the statuses of custom resources,
+so including such one-time resources in a backup can result in an unnecessary reissuance
+after a restore as without the status fields `cert-manager` will not be able to tell that,
+for example, an `Order` has already been fulfilled.
+To avoid unnecessary reissuance, we recommend that `Order`s and `Challenge`s are excluded
+from the backup. We also don't recommend backing up `CertificateRequest`s, see [Backing up CertificateRequests](#backing-up-certificaterequests)
```
Comment (wallrj) [inline content/en/docs/tutorials/backup.md:97]: That's a nice explanation.

**Review by wallrj — CHANGES_REQUESTED** [review]
Thanks for adding all that extra detail.

* I added a few suggestions
* I'd prefer the use of `(...)` or `—` instead of `- `, and in some of those cases, separate sentences might be easier to read.

File: .spelling [hunk 4]
```diff
@@ -139,6 +139,7 @@ propagations
 publicised
 RBAC
 rebase
+reissuance
```
[the remark this answers is not in the document]
Comment (irbekrm) [reply .spelling:142]: Thanks- I've changed `re-issue` -> `reissue` for consistency.

**Comment by irbekrm** [comment]
Thanks for the review @wallrj - I've added updated the PR. Thanks for looking at the sentences, I did feel like they were rambling a bit, so was hoping someone could suggest some improvements.

File: content/en/docs/tutorials/backup.md [hunk 5]
```diff
@@ -49,5 +57,109 @@ created above after installing cert-manager.
...
+are using [`ingress-shim`](../../usage/ingress/).
+
+#### Excluding some cert-manager resources from backup
+
+`cert-manager` has a number of resources- `Order`s, `Challenge`s and
+`CertificateRequest`s -that are designed to represent a point-in-time operation
+such as a request for a certificate. As such, their status often depends on other
+ephemeral resources (i.e a temporary Secret holding a private key), so
+`cert-manager` cannot always correctly re-create the status of these resources
+by just looking at the cluster state at some later time.
```
Comment (wallrj) [inline content/en/docs/tutorials/backup.md:85]:
```suggestion
`cert-manager` has a number of resources — `Order`s, `Challenge`s and
`CertificateRequest`s  — that are designed to represent a point-in-time operation
such as a request for a certificate. As such, their status often depends on other
ephemeral resources (i.e a temporary Secret holding a private key), so
`cert-manager` cannot always correctly re-create the status of these resources
by just looking at the cluster state at some later time.
```

**Review by wallrj — CHANGES_REQUESTED** [review]
I should have highlighted it in the previous review, but there's a lingering dash which should be an emdash, or bracketed.

File: content/en/docs/tutorials/backup.md (hunk 5 above)
Comment (irbekrm) [reply content/en/docs/tutorials/backup.md]: Thanks @wallrj ! I have eliminated those dashes by re-writing this using shorter separate sentences here and in a few more places 😄

**Review by wallrj — APPROVED** [review]
/lgtm
/approve
